File: 2016-08-11-WebViewJavascriptInterafce/src/main/java/tech/thdev/webviewjavascriptinterface/webkit/CustomJavaScript.java
package tech.thdev.webviewjavascriptinterface.webkit;

import android.webkit.JavascriptInterface;

import tech.thdev.webviewjavascriptinterface.webkit.listener.OnCustomJavaScriptListener;

/**
 * Created by Tae-hwan on 8/8/16.
 */

public class CustomJavaScript {

    private OnCustomJavaScriptListener listener;

    public CustomJavaScript() {

    }

    public void setOnCustomJavascriptListener(OnCustomJavaScriptListener listener) {
        this.listener = listener;
    }

    @JavascriptInterface
    public void updateKeyword(String keyword) {
        if (listener != null) {
            listener.onUpdateKeyword(keyword);
        }
    }

    @JavascriptInterface
    public void changeWebView(String url) {
        if (listener != null) {
            listener.onChangeWebView(url);
        }
    }

    @JavascriptInterface
    public void hideKeyboard() {
        if (listener != null) {
            listener.hideKeyboard();
        }
    }
}

File: app/src/main/java/tech/thdev/app/webkit/CustomJavaScript.kt
package tech.thdev.app.webkit

import android.webkit.JavascriptInterface
import tech.thdev.app.webkit.listener.OnCustomJavaScriptListener

/**
 * Created by Tae-hwan on 8/8/16.
 */
class CustomJavaScript {

    private var listener: OnCustomJavaScriptListener? = null

    fun setOnCustomJavascriptListener(listener: OnCustomJavaScriptListener) {
        this.listener = listener
    }

    @JavascriptInterface
    fun updateKeyword(keyword: String) {
        listener?.onUpdateKeyword(keyword)
    }

    @JavascriptInterface
    fun changeWebView(url: String) {
        listener?.onChangeWebView(url)
    }

    @JavascriptInterface
    fun hideKeyboard() {
        listener?.hideKeyboard()
    }
}